components/cu_transform/src/broadcaster.rs

```diff
@@ -0,0 +1,277 @@
+
+        // Encode each transform
+        for transform in &self.transforms {
+            // Encode transform.transform (Transform3D)
+            bincode::enc::Encode::encode(&transform.transform, encoder)?;
+
+            // Encode timestamp
+            bincode::enc::Encode::encode(&transform.stamp, encoder)?;
+
+            // Encode parent_frame string
+            bincode::enc::Encode::encode(&transform.parent_frame, encoder)?;
+
+            // Encode child_frame string
+            bincode::enc::Encode::encode(&transform.child_frame, encoder)?;
+        }
+
+        Ok(())
+    }
+}
+
+impl<T: Copy + Debug + Default + bincode::enc::Encode + bincode::de::Decode<()> + 'static>
+    bincode::de::Decode<()> for TransformBroadcastPayload<T>
+{
+    fn decode<D: bincode::de::Decoder<Context = ()>>(
+        decoder: &mut D,
+    ) -> Result<Self, bincode::error::DecodeError> {
+        // Decode the number of transforms
+        let len = <usize as bincode::de::Decode<()>>::decode(decoder)?;
+
+        // Create vector to hold the transforms
+        let mut transforms = Vec::with_capacity(len);
+
+        // Decode each transform
+        for _ in 0..len {
+            // Decode transform.transform (Transform3D)
+            let transform_3d = <Transform3D<T> as bincode::de::Decode<()>>::decode(decoder)?;
+
+            // Decode timestamp
+            let stamp = <CuTime as bincode::de::Decode<()>>::decode(decoder)?;
+
+            // Decode parent_frame string
+            let parent_frame = <String as bincode::de::Decode<()>>::decode(decoder)?;
+
+            // Decode child_frame string
+            let child_frame = <String as bincode::de::Decode<()>>::decode(decoder)?;
+
+            // Create StampedTransform and add to vector
+            transforms.push(StampedTransform {
+                transform: transform_3d,
+                stamp,
+                parent_frame,
+                child_frame,
+            });
+        }
+
+        Ok(Self { transforms })
+    }
+}
+
+pub struct TransformBroadcaster<'cl, T> {
+    _marker: PhantomData<&'cl T>,
+}
+
+impl<T> Freezable for TransformBroadcaster<'_, T> {
+    fn freeze<E: bincode::enc::Encoder>(
+        &self,
+        _encoder: &mut E,
+    ) -> Result<(), bincode::error::EncodeError> {
+        Ok(())
+    }
+
+    fn thaw<D: bincode::de::Decoder>(
+        &mut self,
+        _decoder: &mut D,
+    ) -> Result<(), bincode::error::DecodeError> {
+        Ok(())
+    }
+}
+
+impl<'cl, T> CuSrcTask<'cl> for TransformBroadcaster<'cl, T>
+where
+    T: CuMsgPayload + Copy + Debug + 'static,
+{
+    type Output = output_msg!('cl, TransformBroadcastPayload<T>);
+
+    fn new(_config: Option<&ComponentConfig>) -> CuResult<Self>
+    where
+        Self: Sized,
+    {
+        Ok(Self {
+            _marker: PhantomData,
+        })
+    }
+
+    fn process(&mut self, clock: &RobotClock, new_msg: Self::Output) -> CuResult<()> {
+        new_msg.metadata.tov = Tov::Time(clock.now());
```

Comment:
yeah, isn't it your time in your Stamped version?
If needed we might want to add a Set of TS in ToV:
  `Range(CuTimeRange)` <- I think this is the best we can do today but it could makes sense to add `Discreet(Array<CuTime>)` for that purpose